packages/two_dimensional_scrollables/lib/src/common/implementation_span.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/gestures.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'span.dart';
+
+// This file is NOT exported as part of the package.
+
+/// A contextual used by the [RenderTableViewport].
+///
+/// Used to track the layout information, and additional configurations of the
+/// span during multiple layout passes.
+abstract class ImplementedSpan
```

Comment:
The name is kinda strange since it doesn't `implements Span`. If there's not a more descriptive name, would BaseSpan be appropriate (as in it is the base class for Spans)?